Now playing screen

Arunabha-Mukhopadhyay opened the pull request:
**Closes #14**

This pull request introduces a new, fully functional full-screen "Now Playing" view to provide users with an immersive experience for the currently playing song. The implementation centralizes all player state in Zustand to ensure the new view is perfectly synchronized with the mini-player.

### Features Implemented

-   **Play/Pause Functionality:** Users can toggle between playing and pausing the current track.
-   **Next & Previous Track Navigation:** Implemented the ability to skip to the next or previous track in the playlist.
-   **Shuffle Mode:** Added shuffle functionality that plays tracks in a random order when enabled.
-   **Loop Mode:** Implemented loop functionality:
    -   *Loop One:* Repeats the current track.
    -   *Loop All:* Repeats the entire playlist.
-   **Seek Bar & Time Tracking:**
    -   Users can drag the seek bar to jump to a specific position in the track.
    -   Displays both **elapsed time** and **total duration** of the track.
-   **Volume Control:** Added volume adjustment via a slider for a better listening experience.
-   **Lyrics Display (In Progress):**
    -   UI integration for displaying lyrics has been added.
    -   *Note: Lyrics fetching and scrolling logic is pending completion.*

### Work in Progress

-   **Lyrics Display:** The UI for displaying lyrics has been integrated. However, **this feature is not functional yet**, as I am currently facing some errors with the lyrics fetching and scrolling logic. This can be addressed in a future update.


### How to Test:

1.  Run the application.
2.  Play any song from the list. The mini-player bar will appear.
3.  Click on the album art or song title in the mini-player.
4.  The full-screen "Now Playing" view will open.
5.  Verify that all implemented features (Play/Pause, Seek, Volume, etc.) are functional and stay in sync between the full-screen view and the mini-player.

### Final Result:
<img width="1440" height="818" alt="Screenshot 2025-10-04 at 1 55 16 AM" src="https://github.com/user-attachments/assets/a34f89a7-abc7-4e63-ae29-25d86bfcfad7" />



```eof

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* New Features
  * Full-screen Now Playing with lyrics, queue, seek bar, volume, shuffle/loop, keyboard shortcuts, and persistent mini-player.
  * Toast notifications for in-app alerts.
  * Search results now sync with the global playback queue.
  * New slowspin animation.

* Refactor
  * Streamlined layout with conditional sidebar, simplified router/layout flow, and centralized playback state.

* Chores
  * Renamed environment variable keys in example.

* Style
  * CSS/utilities cleanup, scrollbar consolidation, and Tailwind configuration overhaul.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Anmol-TheDev commented on .env.example:
```diff
@@ -1,6 +0,0 @@
-VITE_API_KEY="your-firebase-api-key"
-VITE_AUTH_DOMAIN="your-firebase-auth-domain"
-VITE_PROJECT_ID="your-firebase-project-id"
-VITE_STORAGE_BUCKET="your-firebase-storage-bucket"
```
why you guys are removing .evn.example, fix this take the latest pull from main according to that define env variables

Anmol-TheDev commented on src/Auth/firebase.js:
```diff
@@ -2,13 +2,13 @@ import { initializeApp } from "firebase/app";
-    appId:import.meta.env.VITE_appId,
-    measurementId:import.meta.env.VITE_measurementId
+    apiKey: import.meta.env.VITE_API_KEY,
+    authDomain: import.meta.env.VITE_AUTH_DOMAIN,
```
change env variables according to new pattern

Anmol-TheDev reviewed (commented):
fix these changes

Anmol-TheDev commented:
Fix the changes I have mentioned and improve the design of this page. Use a subtle gradient like YouTube already uses, function color generator, take reference from this, and improve the queue section 
<img width="3179" height="1722" alt="image" src="https://github.com/user-attachments/assets/6bf92272-36f5-4340-91f6-bac973f25aca" />

Arunabha-Mukhopadhyay commented:
Hi @Anmol-TheDev , I've made the requested changes. Please take another look when you have a moment!

Arunabha-Mukhopadhyay commented:
This pull request introduces a comprehensive set of features to establish the core functionality of the music player application. It includes global state management with Zustand, integration with the Saavn API, and a fully synchronized playback UI with both a mini and a full-screen player.

Core Features Implemented
Global Music Playback & UI
Implemented global music playback using ReactPlayer.

Created two synchronized player views:

A mini spinning player for persistent, site-wide control.

A fullscreen "Now Playing" screen for an immersive listening experience.

State Management with Zustand
Centralized all application state using Zustand for reliability and consistency.

Manages key global states, including:

Playback queue and the current song.

Player controls: volume, shuffle, and loop modes.

UI state: fullscreen toggle and lyrics visibility.

Search & API Integration
Integrated the Saavn API to dynamically fetch songs, albums, and artists.

Search results are seamlessly synchronized with the global playback queue.

UI/UX Enhancements
Volume Persistence: The user's volume setting is saved in local storage and remembered across sessions.

Custom Scrollbars: Styled the application's scrollbars for a more polished and consistent aesthetic.

Dark Mode Support: Implemented a dark theme using CSS variables and Tailwind for a modern look.

How to Test
Launch the application and use the search bar.

Play a song from the results to activate the mini-player.

Test the mini-player controls (play/pause, skip).

Click the mini-player to open the full-screen "Now Playing" view.

Verify all controls (volume, seek, shuffle, loop) are functional and synchronized between both views.

Final Result:
<img width="1440" height="818" alt="Screenshot 2025-10-04 at 1 42 33 PM" src="https://github.com/user-attachments/assets/648bef87-4097-46bc-89ee-47b1d3dcf767" />

Arunabha-Mukhopadhyay commented:
added the .env.example file
